src/core/net/srp_client.cpp

```diff
@@ -1428,6 +1460,75 @@ void Client::HandleTimer(void)
     }
 }
 
+#if OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_API_ENABLE
+
+void Client::EnableAutoStartMode(AutoStartCallback aCallback, void *aContext)
+{
+    mAutoStartCallback = aCallback;
+    mAutoStartContext  = aContext;
+
+    VerifyOrExit(!mAutoStartModeEnabled);
+    mAutoStartModeEnabled = true;
+    CheckAutoStart();
+
+exit:
+    return;
+}
+
+void Client::CheckAutoStart(void)
```

Comment:
nit: the method name sounds like it won't change any states. May rename to names like `UpdateServerForAutoStart`?

Reply:
This method is checking whether we need to auto-start or stop  and is called whenever there is a event/change.  I will rename to `ProessAutoStart()`.